queue/memory.go

```diff
@@ -120,9 +120,12 @@ func (a *memoryAck) Reject(requeue bool) error {
 // Next returns the next job in the iter.
 func (i *memoryJobIter) Next() (*Job, error) {
 	for {
+		i.RLock()
 		if i.closed {
+			i.RUnlock()
 			return nil, ErrAlreadyClosed
 		}
+		i.RUnlock()
```

Comment:
Two questions:

1. If the only purpose of this lock is check for the `closed` attribute, maybe it's better just have an `isClosed` method which does the `Lock` and `Unlock` to avoid having this mixed in the code with several levels to `Unlock`.

2. If we are checking if the `memoryJobIter` is closed to perfom some other task, what would happen if before run the next instruction someone else close it? Are we sure this is the only part of the code we need to lock?

Reply:
1) yes, much better
2) because of the lock, everything is synchronized, so if someone closes it, there's no next. Yes, only the access to the properties need to be locked